edi_webservice_oca/tests/test_send_saving_response.py

```diff
@@ -0,0 +1,207 @@
+    can't use the swapping method.
+
+    Choice have to be made at developement level to decide what to do according
+    the current record type
+    """
+
+    @classmethod
+    def _setup_records(cls):
+        super()._setup_records()
+        cls.loader = FakeModelLoader(cls.env, cls.__module__)
+        cls.addClassCleanup(cls.loader.restore_registry)
+        cls.loader.backup_registry()
+        cls.edi_exchange_state_changed = False
+
+        class EDIExchangeRecord(models.Model):
+            _name = "edi.exchange.record"
+            _inherit = "edi.exchange.record"
+
+            def _save_ws_response(self, content, status_code):
+                super()._save_ws_response(content, status_code)
+                if (
+                    self.backend_id.webservice_backend_id.save_response
+                    and status_code != 200
+                ):
+                    self.edi_exchange_state = "output_error_on_send"
+                    # because we are in unittest we have only one transaction
+                    # as we are creating the exchange record in this transaction
+                    # we can't really instantiate a new transaction
+                    # So this edi_exchange_state get overwritten by
+                    # the edi_oca.models.edi_backend.EDIBackend.exchange_send
+                    # finally clause that would be rollback in production server
+                    # so use this attribute to assert state has been changed
+                    cls.edi_exchange_state_changed = True
+
+        cls.loader.update_registry((EDIExchangeRecord,))
+
+        cls.ws_backend = cls.backend.webservice_backend_id
+        cls.ws_backend.save_response = True
+        cls.settings = """
+        components:
+          send:
+            usage: webservice.send
+            work_ctx:
+              webservice:
+                method: post
+                kwargs:
+                  url_params:
+                    endpoint: push/here
+        """
+        cls.record.edi_exchange_state = "output_pending"
+
+        @contextmanager
+        def _consumer_record_no_new_env(self, record, new_cursor=True):
+            yield record
+
+        cls._consumer_record_no_new_env = _consumer_record_no_new_env
+
+    @responses.activate
+    def test_component_send_save_results_on_http_error(self):
+        self.record.type_id.set_settings(self.settings)
+        url = "https://foo.test/push/here"
+        response = (
+            "{'error': 'something that details what went wrong to help investigations'}"
+        )
+        responses.add(
+            responses.POST,
+            url,
+            body=response,
+            status=401,
+        )
+        with mock.patch(
+            "odoo.addons.webservice.models.webservice_backend."
+            "WebserviceBackend._consumer_record_env",
+            side_effect=self._consumer_record_no_new_env,
+        ) as mo:
+            with self.assertRaisesRegex(HTTPError, "Unauthorized"):
```

Comment:
Hi @simahawk,

I've rebase this PR and get new features :). So I found that [this PR](https://github.com/OCA/edi/pull/796/files#diff-6252f8654b549013124ad45c71930bd855b1ea7ad31ec323d546516e494ee7ccR354) add `OSError` as `retryable_exceptions`.

Are you aware that [`HTTPError` inherits from `OSError`](https://github.com/python/cpython/blob/3.12/Lib/urllib/error.py) so that all kind of HTTP Errors (3xx, 4xx, 5xx) are retried ? And is that a desired side effect ?